quickwit/quickwit-indexing/src/actors/uploader.rs

```diff
@@ -295,29 +295,63 @@ impl Handler<PackagedSplitBatch> for Uploader {
         tokio::spawn(
             async move {
                 fail_point!("uploader:intask:before");
-                let mut packaged_splits_and_metadatas = Vec::new();
-                for split in batch.splits {
+
+                let mut split_metadata_list = Vec::with_capacity(batch.splits.len());
+                for packaged_split in batch.splits.iter() {
                     if batch.publish_lock.is_dead() {
                         // TODO: Remove the junk right away?
                         info!("Splits' publish lock is dead.");
                         split_update_sender.discard()?;
                         return Ok(());
                     }
-                    let upload_result = stage_and_upload_split(
-                        &split,
+
+                    let split_streamer = SplitPayloadBuilder::get_split_payload(
+                        &packaged_split.split_files,
+                        &packaged_split.hotcache_bytes,
+                    )?;
+                    let split_metadata = create_split_metadata(
+                        &packaged_split.split_attrs,
+                        packaged_split.tags.clone(),
+                        split_streamer.footer_range.start as u64..split_streamer.footer_range.end as u64,
+                    );
+
+                    split_metadata_list.push(split_metadata);
+                }
+
+                metastore
+                    .stage_splits(&index_id, split_metadata_list.clone())
+                    .instrument(tracing::info_span!("stage_splits"))
```

Comment:
I don't think we need this instrument.
It is already in the metastore implementation as an annotation.

```suggestion
```